examples/webgl_custom_clipping.html

```diff
@@ -0,0 +1,399 @@
+					mesh.matrixAutoUpdate = false;
+
+					volumeVisualization.add( mesh );
+
+				}
+
+				scene.add( volumeVisualization );
+
+
+				var ground = new THREE.Mesh( planeGeometry,
+						new THREE.MeshPhongMaterial( {
+							color: 0xa0adaf, shininess: 150 } ) );
+				ground.rotation.x = - Math.PI / 2;
+				ground.scale.multiplyScalar( 3 );
+				ground.receiveShadow = true;
+				scene.add( ground );
+
+				// Renderer
+
+				var container = document.body;
+
+				renderer = new THREE.WebGLRenderer();
+				renderer.shadowMap.enabled = true;
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( window.innerWidth, window.innerHeight );
+				window.addEventListener( 'resize', onWindowResize, false );
+				container.appendChild( renderer.domElement );
+				// Clipping setup:
+				globalClippingPlanes = createPlanes( GlobalClippingPlanes.length );
+				renderer.clippingPlanes = Empty;
+				renderer.localClippingEnabled = true;
+
+				// Stats
+
+				stats = new Stats();
+				container.appendChild( stats.domElement );
+
+				// Controls
+
+				var controls = new THREE.OrbitControls( camera, renderer.domElement );
+				controls.target.set( 0, 1, 0 );
+				controls.update();
+
+				// GUI
+
+				var gui = new dat.GUI(),
+					folder = gui.addFolder( "Local Clipping" ),
+					props = {
+						get 'Enabled'() { return renderer.localClippingEnabled; },
+						set 'Enabled'( v ) {
+								renderer.localClippingEnabled = v;
+								if ( ! v ) volumeVisualization.visible = false; },
+
+						get 'Shadows'() { return clipMaterial.clipShadows; },
+						set 'Shadows'( v ) { clipMaterial.clipShadows = v; },
+
+						get 'Visualize'() { return volumeVisualization.visible; },
+						set 'Visualize'( v ) {
+								if ( renderer.localClippingEnabled )
+									volumeVisualization.visible = v; }
+					};
+
+				folder.add( props, 'Enabled' );
+				folder.add( props, 'Shadows' );
+				folder.add( props, 'Visualize' ).listen();
+
+				gui.addFolder( "Global Clipping" ).
+						add( {
+							get 'Enabled'() { return renderer.clippingPlanes !== Empty; },
+							set 'Enabled'( v  ) { renderer.clippingPlanes = v ?
+									globalClippingPlanes : Empty; }
+						}, "Enabled" );
+
+
+				// Start
+
+				startTime = Date.now();
+
+			}
+
+			function onWindowResize() {
+
+				camera.aspect = window.innerWidth / window.innerHeight;
+				camera.updateProjectionMatrix();
+
+				renderer.setSize( window.innerWidth, window.innerHeight );
+
+			}
+
+			var transform = new THREE.Matrix4(),
+				tmpMatrix = new THREE.Matrix4();
+
+			function animate() {
+
+				var currentTime = Date.now(),
+					time = ( currentTime - startTime ) / 1000;
+
+				requestAnimationFrame( animate );
+
+				transform.makeTranslation( 0, 1, 0 ).
+						multiply( tmpMatrix.makeRotationX( time * 0.5 ) ).
+						multiply( tmpMatrix.makeRotationY( time * 0.2 ) );
+
+				object.matrix.copy( transform );
+				object.updateMatrixWorld( true );
```

Comment:
We had many debates about what the user should and should-not be able to do years ago, so there is a lot of history regarding this issue. New users often don't know what they are doing. The policy has been to advise against setting the object matrix directly. That is the renderer's job.

That being said, I have no problem revisiting this issue if you feel you have a compelling reason to do so.